Framework/PythonInterface/mantid/geometry/src/Exports/CSGObject.cpp

```diff
@@ -5,18 +5,52 @@
 //   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 // SPDX - License - Identifier: GPL - 3.0 +
 #include "MantidGeometry/Objects/CSGObject.h"
+#include "MantidGeometry/Rendering/GeometryTriangulator.h"
 #include "MantidPythonInterface/core/GetPointer.h"
 #include <boost/python/class.hpp>
 #include <boost/python/copy_const_reference.hpp>
 #include <boost/python/register_ptr_to_python.hpp>
 
+#include "MantidPythonInterface/core/Converters/NDArrayTypeIndex.h"
+#include "MantidPythonInterface/core/Converters/WrapWithNDArray.h"
+#include "MantidPythonInterface/core/GetPointer.h"
+
+#include <boost/python/register_ptr_to_python.hpp>
+#include <boost/python/self.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+#define PY_ARRAY_UNIQUE_SYMBOL GEOMETRY_ARRAY_API
+#define NO_IMPORT_ARRAY
+#include <numpy/arrayobject.h>
+
+using Mantid::Geometry::IObject;
+// using Mantid::Geometry::MeshObject;
+using namespace Mantid::PythonInterface::Converters;
+using namespace boost::python;
 using Mantid::Geometry::BoundingBox;
 using Mantid::Geometry::CSGObject;
 using Mantid::Geometry::IObject;
 using namespace boost::python;
 
 GET_POINTER_SPECIALIZATION(CSGObject)
 
+boost::python::object wrapMeshWithNDArray(CSGObject &self) {
+  // PyArray_SimpleNewFromData doesn't interact well with smart pointers so use raw pointer
+
+  auto vertices = self.getTriangleVertices();
+  auto triangles = self.getTriangleFaces();
+  size_t numberTriangles = triangles.size() / 3;
```

Comment:
CSGObject has a method to find the number of triangles right?

Reply:
It does and I just tried it, but it doesn't work. I gets blocked by a canTriangulate() call (L68 in GeometryHandler.cpp) which assumes False for a CSGObject.

Reply:
I've now used `self.numberOfTriangles()`